### ISSUE TITLE
Milestone 3 

### ISSUE THREAD

**Issue by sean-m-mckay** [UBC-MDS/wine-origin-prediction#28]
To turn into a script
- All logic from the report
- The scripts should call the modules in the source folder as needed
- They should output to disk, not to memory so they are self-contained

Script input and output table

| Script Name | Input Params | purpose | Notes | Assignee
|---|---|---|---|---|
| read data | path to output file to | read data and save to file | data is in the pip package, so no need to specify the URL to read from | Hina
| preprocessing | path to input train data, path to input test data, output file path, output preprocessor (Pickle) | preprocess data | | Yimeng
| eda | path to input processed data file, path to save plot to (figure) | plot data analysis | | Sean
| model optimization | path to input training data file, path for preprocessor (Pickle), path to output model to (Pickle), path to output CV results (figure) | train and optimize the model | | Julia
| evaluation | path to test data, path to model (Pickle), path to output results to (CSV) | evaluate model on a test set | | Hina

**Comment by hbandukw** [comment]
For 'ead data' - I am including an additional input  i.e.  "ucirepo_id"

**Comment by sean-m-mckay** [comment]
@hbandukw I think it's better if we hardcode in this case because we don't know if it will work for other uci datasets  (and the rest of the analysis steps will fail for other datasets) - or if you leave it, might want to add a comment that it's only tested on this one dataset specifically
